packages/playwright-ct-react/registerSource.mjs

```diff
@@ -77,17 +79,39 @@ window.playwrightMount = async (component, rootElement, hooksConfig) => {
       App = () => wrapper;
   }
 
-  ReactDOM.render(App(), rootElement);
+  if (rootRegistry.has(rootElement)) {
+    throw new Error(
+        'Attempting to mount a component into an container that already has a React root. ' +
+        'Either unmount the React root first with `playwrightUnmount(rootElement)` or update the root instead with `playwrightUpdate(rootElement, component)`'
+    );
+  }
+
+  const root = createRoot(rootElement);
+  rootRegistry.set(rootElement, root);
+  // TODO: Do we need to flush sync?
+  root.render(App());
 
   for (const hook of /** @type {any} */(window).__pw_hooks_after_mount || [])
     await hook({ hooksConfig });
 };
 
 window.playwrightUnmount = async rootElement => {
-  if (!ReactDOM.unmountComponentAtNode(rootElement))
+  const root = rootRegistry.get(rootElement);
+  if (root === undefined)
     throw new Error('Component was not mounted');
+
+  // TODO: Do we need to flush sync?
```

Comment:
You tell us!

Reply:
Yeah I'm not sure yet we need to. It kind of depends on the use case for the mount hooks. Because even in React 17 the mount hooks might run before every `useEffect` effect was executed.

Problem with `flushSync` is that we change how the UI might be seen by actual users.

I think about it again. Maybe it's better to not flush sync by default but expose another method/option to do so.

Reply:
Not going to flushSync for now and see what the specific use cases are. Effects were already not flushed sync.